.github/workflows/codeql.yml

```diff
@@ -1,8 +1,10 @@
 name: CodeQL Analysis
 
```

Comment:
> For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork.

Definately think it's worth trying `permissions: {}` at the start of the file  to avoid overly-permissive actions